mergify_engine/actions/utils.py

```diff
@@ -45,19 +47,30 @@ async def validate_bot_account(
             ),
         )
 
-    if need_write_permission:
+    if required_permissions is None:
+        required_permissions = ["admin", "write", "maintain"]
+
+    if required_permissions:
```

Comment:
This is suboptimal, as if `required_permissions` is `None` (e.g., `comment`), then there's no need to check for permissions at all.

Reply:
comment has `[]` not `None` to avoid the permission lookup. The code is confusing, just because mypy complains I was using mutable in function arguments, so I put `None`. It may be more readable with an `Unset` object marker.

Reply:
Right. Indeed it's a bit confusing and I understand mypy. 😆